### PR TITLE
Dealership class

### PR DESCRIPTION
**Pull request by CwongM**
Provided constructors for point 3 and 4 of Group Assignment document
Provided an additional dealer constructor for non-existent dealers that have yet to be added
Methods in Dealership class fulfill point 5 and 6 of Group Assignment document

### CODE REVIEW

**Comment by CwongM** [comment]
Awaiting approval from at least 2 group members before merging
EDIT: Waiting until September 14th meeting before considering merge

File: src/Vehicle.java
Comment (tomatogrove) [inline src/Vehicle.java]: just a reminder that Vehicle was assigned to Iab

File: src/Dealership.java
Comment (tomatogrove) [inline src/Dealership.java]: I think in class we talked about how a map might be better since it would be easier to access/manipulate the vehicles based on vehicleID rather than having to implement a search method.

With how it is now there may be multiple vehicles with the same vehicleID which defeats the purpose of the ID. 

You could have something like:

private Map<Integer,Vehicle> vehicleInventory;

and then in the constructor initialize it 
this.vehicleInventory= new HashMap<>();

javadoc for Map: https://docs.oracle.com/javase/8/docs/api/java/util/Map.html?is-external=true

File: src/Dealership.java
Comment (tomatogrove) [inline src/Dealership.java:64]: It would be helpful to the DealerGroup to implement a getVehicleInventory() method.

File: src/Vehicle.java
Comment (CwongM) [reply src/Vehicle.java]: Thanks, I'll remove my iteration

File: src/Dealership.java
Comment (CwongM) [reply src/Dealership.java:64]: Implementing a getter method for that field

File: src/Dealership.java
Comment (CwongM) [reply src/Dealership.java]: Thanks for the reminder, I'll make adjustments to use maps instead with an integer key to start with.

**Comment by CwongM** [comment]
Awaiting additional feedback and reviewer to approve this pull request

File: src/Dealership.java
Comment (tomatogrove) [inline src/Dealership.java:56]: The class looks good I would just recommend changing the enable and disable methods. They aren't static methods so i would just recommend changing it from 
dealer.vehicleAcquisition 
to 
this.vehicleAcquisition

File: src/Dealership.java
Comment (tomatogrove) [inline src/Dealership.java:63]: Also here it might be good to just play around with the formatting of this string you're sending back. maybe just test this out and see how it looks.

File: src/Dealership.java
Comment (CwongM) [reply src/Dealership.java:56]: Adjusted and removed parameter from methods mentioned above and the inventory method

File: src/Dealership.java
Comment (CwongM) [reply src/Dealership.java:63]: Screenshot 2022-09-16 172410
Output is as expected. Could use some refinement in the toString method of the Vehicle class.

**Comment by CwongM** [comment]
Also updated the addIncomingVehicle method to restrict vehicle type and prevent duplicate keys from overwriting vehicles in a dealer's inventory.

**Review by tomatogrove — APPROVED** [review]

**Pull request merged**
